src/Tasks/GenerateResource.cs

```diff
@@ -3539,22 +3498,14 @@ public static bool TryCreateCodeDomProvider(TaskLoggingHelper logger, string str
             {
                 provider = CodeDomProvider.CreateProvider(stronglyTypedLanguage);
             }
-#if FEATURE_SYSTEM_CONFIGURATION
-            catch (ConfigurationException e)
-            {
-                logger.LogErrorWithCodeFromResources("GenerateResource.STRCodeDomProviderFailed", stronglyTypedLanguage, e.Message);
-                return false;
-            }
-#endif
-            catch (SecurityException e)
+            catch (Exception e) when (e is ConfigurationException || e is SystemException)
```

Comment:
Previously this was ConfigurationException and SecurityException.  Now ConfigurationException and SystemException.  Why?  Also, are you actually seeing the ConfigurationException on .NETCore?

Reply:
No, as I recall the only exception I could catch in the test that exercised this codepath was a `SystemException`-derived private type. Since you're discouraging using ConfigurationManager, I'll leave the full framework type catches alone and catch that only for core.

Reply:
Details for posterity: when passing an unrecognized language (like in the test `Microsoft.Build.UnitTests.GenerateResource_Tests.InProc.PropertyErrors.UnknownStronglyTypedLanguage`), .NET Core CodeDOM [throws](https://github.com/dotnet/corefx/blob/2fac89f63c644ab81d936d62c91e6003302e718d/src/System.CodeDom/src/System/CodeDom/Compiler/CodeDomProvider.cs#L84) a [private `ConfigurationErrorsException`](https://github.com/dotnet/corefx/blob/2fac89f63c644ab81d936d62c91e6003302e718d/src/System.CodeDom/src/System/CodeDom/Compiler/CodeDomProvider.cs#L215-L222), which is a `SystemException`.